hooks/collect-metrics

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+
+# Load modules from $JUJU_CHARM_DIR/lib.
+import sys
+sys.path.append('lib')
+
+# Activate the virtualenv.
+from charms.layer.basic import activate_venv  # noqa: E402
+activate_venv()
+
+# Start the script as usual.
+import ssl  # noqa: E402
+import subprocess  # noqa: E402
+from urllib import request  # noqa: E402
+
+from firestealer import text_to_samples  # noqa: E402
+import yaml  # noqa: E402
+
+
+# Map metrics names to samples to be collected.
+METRICS = {
```

Comment:
I considered maybe deriving this from `metrics.yaml` directly but maybe that would be too inflexible later on...

Reply:
Yeah, I considered the same, but not really worth it for now, and the rule for deriving values feels incidental.